pkg/shared/utils/arrays/generic_array_utils_test.go

```diff
@@ -1,33 +1,75 @@
 package arrays
 
 import (
-	. "github.com/onsi/gomega"
 	"testing"
+
+	. "github.com/onsi/gomega"
 )
 
-func TestGenericFindFirstWithStrings(t *testing.T) {
-	RegisterTestingT(t)
-	idx, val := FindFirst(func(x interface{}) bool { return x.(string) == "Red" }, "This", "Is", "Red", "Hat")
-	Expect(idx).To(Equal(2))
-	Expect(val).To(Equal("Red"))
-}
+func Test_FindFirst(t *testing.T) {
+	type args struct {
+		predicate func(x interface{}) bool
+		values    []interface{}
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  int
+		want1 interface{}
+	}{
+		{
+			name: "Return -1 index and nil if no match",
+			args: args{
+				predicate: func(x interface{}) bool { return x.(int) > 5 },
+				values:    []interface{}{1, 2, 3, 4},
+			},
+			want:  -1,
+			want1: nil,
+		},
+		{
+			name: "Return with index and matching integer",
+			args: args{
+				predicate: func(x interface{}) bool { return x.(int) > 5 },
+				values:    []interface{}{1, 2, 3, 4, 5, 6, 7, 8},
+			},
+			want:  5,
+			want1: 6,
+		},
+		{
+			name: "Return with index and matching string",
+			args: args{
+				predicate: func(x interface{}) bool { return x.(string) == "Red" },
+				values:    []interface{}{"This", "Is", "Red", "Hat"},
+			},
+			want:  2,
+			want1: "Red",
+		},
+		{
+			name: "Return with index and matching incompatible types",
+			args: args{
+				predicate: func(x interface{}) bool {
+					if val, ok := x.(int); ok {
+						return val > 5
+					} else {
+						return false
+					}
+				},
+				values: []interface{}{1, 2, "Hello", 4, 5, 6, 7, 8},
+			},
+			want:  5,
+			want1: 6,
+		},
+	}
 
-func TestGenericFindFirstWithInt(t *testing.T) {
 	RegisterTestingT(t)
-	idx, val := FindFirst(func(x interface{}) bool { return x.(int) > 5 }, 1, 2, 3, 4, 5, 6, 7, 8)
-	Expect(idx).To(Equal(5))
-	Expect(val).To(Equal(6))
-}
 
-func TestGenericWithIncompatibleTypes(t *testing.T) {
-	RegisterTestingT(t)
-	idx, val := FindFirst(func(x interface{}) bool {
-		if val, ok := x.(int); ok {
-			return val > 5
-		} else {
-			return false
-		}
-	}, 1, 2, "Hello", 4, 5, 6, 7, 8)
-	Expect(idx).To(Equal(5))
-	Expect(val).To(Equal(6))
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotIndex, gotVal := FindFirst(tt.args.predicate, tt.args.values...)
+			Expect(gotIndex).To(Equal(tt.want))
+			if gotVal != nil {
+				Expect(gotVal).To(Equal(tt.want1))
+			}
```

Comment:
What if `gotVal` is `nil` but a value was expected? With this test, the case is skipped.
I suggest the change below:
```suggestion
	Expect(gotVal).To(Equal(tt.want1))
```